### PR TITLE
set pending state to deployment events

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/giantswarm/issues/1616. This sets the pending state to deployment events once the TPO is updated. The operator will set the status to success or failure once the deployment is processed.

### CODE REVIEW

File: service/eventer/github/client.go [hunk 1]
```diff
@@ -178,3 +179,45 @@ func (e *Eventer) fetchDeploymentStatus(project string, deployment deployment) (
 
 	return deploymentStatuses, nil
 }
+
+// postDeploymentStatus posts a Deployment Status for the given Deployment.
+func (e *Eventer) postDeploymentStatus(project string, id int, state deploymentStatusState) error {
```
Comment (xh3b4sd) [inline service/eventer/github/client.go:184]: This is all from draughtsman.

File: service/informer/informer.go [hunk 2]
```diff
@@ -166,15 +166,19 @@ func (s *Service) bootWithError() error {
...
+			// deployment status accordingly.
+			err = s.tpo.Ensure(TPO)
+			if err != nil {
+				return microerror.Mask(err)
+			}
+
+			err = s.eventer.SetPendingStatus(d)
+			if err != nil {
+				return microerror.Mask(err)
+			}
```
Comment (xh3b4sd) [inline service/informer/informer.go:181]: We set the pending state when initializing the TPO.

File: service/informer/informer.go [hunk 3]
```diff
@@ -203,6 +207,11 @@ func (s *Service) bootWithError() error {
 			if err != nil {
 				return microerror.Mask(err)
 			}
+
+			err = s.eventer.SetPendingStatus(d)
+			if err != nil {
+				return microerror.Mask(err)
+			}
```
Comment (xh3b4sd) [inline service/informer/informer.go:214]: we set the pending state when continuously updating the TPO.

File: service/informer/informer_test.go [hunk 4]
```diff
@@ -460,3 +469,7 @@ func (e *testEventer) FetchLatest(project, environment string) (eventerspec.Depl
 
 	return eventerspec.DeploymentEvent{}, eventer.NotFoundError
 }
+
+func (e *testEventer) SetPendingStatus(event eventerspec.DeploymentEvent) error {
+	return nil
```
Comment (xh3b4sd) [inline service/informer/informer_test.go:474]: Due to time pressure I did not write tests for this now. Something for boyscouting later.

**Review by rossf7 — APPROVED** [review]

**Pull request merged**